core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -297,6 +297,17 @@ class SnappySessionState(snappySession: SnappySession)
 
   def getTablePartitions(region: CacheDistributionAdvisee): Array[Partition] =
     StoreUtils.getPartitionsReplicatedTable(snappySession, region)
+
+  var isPreparePhase: Boolean = false
```

Comment:
Changed to Option[ParamterValueSet] however could not remove var. Also both the parameters are needed since execution case do not always have pvs but only in prep stmt case.